### PR TITLE
Update CI/CD Automation For Releases

### PR DESCRIPTION
**Pull request by neelmistry94**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'

### Summary of changes
This PR introduces a script that will automate updates to the Package Manifest and Podspec. Turf and MME have been omitted because those are version locked for now. The PR also adds this to our CI process by making the change on CI and pushing a commit to the release branch with the changes. This allows the releaser to test the updates before tagging and merging back to main.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -377,6 +377,11 @@ jobs:
                 command: |
                     ./scripts/release/create-api-downloads-pr.sh mobile-maps-ios ${VERSION}
                 when: always
+            - run:
+                name: Update Package Manifest & Podspec
+                command: |
+                    ./scripts/release/update-spm-manifest.sh ${VERSION} "$(cat MapboxMaps.xcframework.zip.checksum)" 10.0.0-beta.8 10.0.0-beta.13
```
Comment (julianrex) [inline .circleci/config.yml:383]: I wonder if there's scope here for including the update script [this PR](https://github.com/mapbox/mapbox-maps-ios/pull/82) introduces.

Also, what are these versions here? MapboxCoreMaps & MapboxCommon?

File: MapboxMaps.podspec [hunk 2]
```diff
@@ -1,9 +1,9 @@
 Pod::Spec.new do |m|
 
-  version = '10.0.0-beta.12'
+  maps_version = '10.0.0-beta.12'
```
Comment (julianrex) [inline MapboxMaps.podspec:3]: With this change, the script in the above mentioned PR might need changing.

File: scripts/release/update-spm-manifest.sh [hunk 3]
```diff
@@ -4,52 +4,34 @@ set -euo pipefail
...
 sed -i '' s/"let checksum = \".*\""/"let checksum = \"${CHECKSUM}\""/ Package.swift
 sed -i '' s/"mapbox-common-ios.git\", .exact(\".*\")"/"mapbox-common-ios.git\", .exact(\"${COMMON_VERSION}\")"/ Package.swift
 sed -i '' s/"mapbox-core-maps-ios.git\", .exact(\".*\")"/"mapbox-core-maps-ios.git\", .exact(\"${CORE_VERSION}\")"/ Package.swift
 
+# Update MapboxMaps.podspec
+sed -i '' s/"maps_version = '.*'"/"maps_version = '${MAPS_VERSION}'"/ MapboxMaps.podspec
+sed -i '' s/"m.dependency 'MapboxCommon', '.*'"/"m.dependency 'MapboxCommon', '${COMMON_VERSION}'"/ MapboxMaps.podspec
+sed -i '' s/"m.dependency 'MapboxCoreMaps', '.*'"/"m.dependency 'MapboxCoreMaps', '${CORE_VERSION}'"/ MapboxMaps.podspec
+
 #
```
Comment (julianrex) [inline scripts/release/update-spm-manifest.sh:32]: Is there any way we could test the validity of these changes, before the commit & push?

File: .circleci/config.yml (hunk 1 above)
Comment (neelmistry94) [reply .circleci/config.yml:383]: I'm going to make a change to use the Cartfile instead.

For your update_version script, I actually posted a comment wondering why you are updating podspec. I don't see why you are doing that.

File: MapboxMaps.podspec (hunk 2 above)
Comment (neelmistry94) [reply MapboxMaps.podspec:3]: This is absolutely necessary because otherwise my `sed` step for podspec is going to update swift_version

**Review by julianrex — APPROVED** [review]

File: scripts/release/update-spm-manifest.sh (hunk 3 above)
Comment (neelmistry94) [reply scripts/release/update-spm-manifest.sh:32]: We are on a release branch, so the last step of our CI is to make these changes in automation and commit/push. 

Part of the process is for the releaser to validate these changes before tagging

File: scripts/release/update-spm-manifest.sh [hunk 4]
```diff
@@ -4,52 +4,34 @@ set -euo pipefail
...
 
 MAPS_VERSION=${1}
-COMMON_VERSION=${2//v/}
-CORE_VERSION=${3//v/}
-CHECKSUM=${4}
-GITHUB_TOKEN=$(./scripts/get_token.js)
+CHECKSUM=${2}
+COMMON_VERSION=`awk -F'MapboxCommon-ios.json" ==' '{ print $2 }' Cartfile`
+CORE_VERSION=`awk -F'MapboxCoreMaps.json" ==' '{ print $2 }' Cartfile`
```
Comment (macdrevx) [inline scripts/release/update-spm-manifest.sh:13]: We should use `$(...)` instead of backticks: http://mywiki.wooledge.org/BashFAQ/082

**Review by macdrevx — APPROVED** [review]

**Pull request merged**